### PR TITLE
add CODEOWNERS file

### LINKED ISSUE #2584: Code Owners
We need a file like this, but with more Node-y people in it:

Source: https://github.com/lukesneeringer/google-cloud-python/blob/5edcd7e6feba48d7ef9cf59010b286f0edf3aa35/.github/CODEOWNERS
```
# Code owners file.
# This file controls who is tagged for review for any given pull request.

# Luke Sneeringer is the default owner for anything not explicitly
# taken by someone else.
*                         @lukesneeringer

# Danny Hermes is the primary author of the original core library, as
# well as most of the testing code.
core/*                    @dhermes
test_utils/*              @dhermes

# Jon Wayne Parrott is the author of the new google.api.core library.
# NOTE: This trumps the previous core/* directive, since the last match is
#       what matters.
core/google/api/core/*    @jonparrott

# Tres Seaver and Tim Swast share responsibility for BigQuery.
bigquery/*                @tswast @tseaver

# Danny Hermes and Tres Seaver co-authored the BigTable library.
bigtable/*                @dhermes @tseaver

# Danny Hermes is the primary maintainer of the Datastore library.
datastore/*               @dhermes

# Bill Prin is the primary maintainer of the Stackdriver libraries.
error_reporting/*         @waprin
monitoring/*              @waprin @supriyagarg @rimey
logging/*                 @waprin @tseaver

# Luke Sneeringer is the primary author of the ML API clients.
language/*                @lukesneeringer
speech/*                  @lukesneeringer
translate/*               @lukesneeringer
videointelligence/*       @lukesneeringer
vision/*                  @lukesneeringer

# Luke Sneeringer is the primary author of Pub/Sub.
pubsub/*                  @lukesneeringer

# Danny Hermes is the primary author of resource manager
resourcemanager/*         @dhermes

# Tim Swast is the primary author of Runtime Config.
runtimeconfig/            @tswast

# Vikas Kedia is the product owner and Tres Seaver the author for Spanner.
spanner/*                 @vkedia @tseaver

# Doug Greiman and Angela Li are the owners of trace.
trace/*                   @liyanhui1228 @duggelz
```

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #2584 
RE: https://help.github.com/articles/about-codeowners

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+docs/*                        @callmehiphop
+scripts/*                     @callmehiphop
+
+# Dave Gramlich and Stephen Sawchuk are the primary authors and share
+# responsibility for most libraries and tests.
+packages/*                    @callmehiphop @stephenplusplus
+system-test/*                 @callmehiphop @stephenplusplus
+test/*                        @callmehiphop @stephenplusplus
+
+# Ali Sheikh is the primary maintainer of the Logging extension libraries.
```
Comment (stephenplusplus) [inline CODEOWNERS]: @ofrobots is there anyone else I should tag for these libraries / is there better wording to use?

Comment (ofrobots) [reply CODEOWNERS]: Can we do Github teams? If so @GoogleCloudPlatform/node-team.

Comment (stephenplusplus) [reply CODEOWNERS]: I think so-- cc @lukesneeringer on if that matches the intent of this file.

Comment (lukesneeringer) [reply CODEOWNERS]: Try it. If it does not work, we can change it. :-)

Comment (stephenplusplus) [reply CODEOWNERS]: Do we want to replace the @ofrobots mention with that team?

Also, I think that means the entirety of that team would be pinged for all PRs that relate to that code. Is that what we want? I don't know how many people are on that team, but I can imagine their getting pretty annoyed with the frequency of pinging on code they didn't write.

Comment (ofrobots) [reply CODEOWNERS]: The team is ultimately responsible for the code, so it is appropriate to ping the team IMO.

Comment (stephenplusplus) [reply CODEOWNERS]: It's cool with me if that's the way this is supposed to go. Out of curiosity, how many people are on that team?

Comment (stephenplusplus) [reply CODEOWNERS]: Also, should that team be in the catch-all as well?

**Pull request merged**